core/src/validator.rs

```diff
@@ -1062,6 +1063,20 @@ impl Validator {
                     )
                 };
 
+            let rpc_completed_slots_service = if !config.rpc_config.full_api {
```

Comment:
I'm tempted to recommend `let rpc_completed_slots_service = config.rpc_config.full_api.then()` here, but maybe it won't work with the additional pubsub conditional you intend to add?